src/librustc/ty/maps/mod.rs

```diff
@@ -119,6 +119,9 @@ define_maps! { <'tcx>
     /// (inferred) variance.
     [] fn variances_of: ItemVariances(DefId) -> Rc<Vec<ty::Variance>>,
 
+    /// Maps from def-id of a type to its (inferred) outlives.
+    [] fn inferred_outlives_of: PredicatesOfItem(DefId) -> Vec<ty::Predicate<'tcx>>,
```

Comment:
I think your problem is that you are re-using this `PredicatesOfItem` dep-node. We need to make an `InferredOutlivesOf` dep-node. This process is not well documented -- it has been actively changing in the last few days! But roughly speaking you need to [add a `DepNode` variant here](https://github.com/rust-lang/rust/blob/d7acd29ad5a49f6123185502d8ca92822c477f55/src/librustc/dep_graph/dep_node.rs#L478) and then add [a line to this big macro here](https://github.com/rust-lang/rust/blob/d7acd29ad5a49f6123185502d8ca92822c477f55/src/librustc/ty/maps/plumbing.rs#L742), essentially copying the model of `PredicatesOfItem`.